src/System.Linq/src/System/Linq/Enumerable.cs

```diff
@@ -2498,6 +2499,337 @@ public static decimal Average<TSource>(this IEnumerable<TSource> source, Func<TS
         {
             return Enumerable.Average(Enumerable.Select(source, selector));
         }
+
+        public static TSource MaxBy<TSource, TKey>(this IEnumerable<TSource> source, Func<TSource, TKey> keySelector)
+        {
+            return source.MaxBy(keySelector, null);
+        }
```

Comment:
There is incidentally a technical reason to favour passing null in cases like this, as it increases the chances of the method falling below the jitter's threshold for inlining if you consistently favour null over property access, because sometimes the few IL instructions can make the difference.